src/Database/Relations/BelongsToMany.php

```diff
@@ -100,6 +100,110 @@ public function create(array $attributes = [], array $pivotData = [], $sessionKe
         return $model;
     }
 
+    /**
+     * @param mixed $id
+     * @param array $attributes
+     * @param bool  $touch
+     */
+    public function attach($id, array $attributes = [], $touch = true)
+    {
+        $insertData = $this->formatAttachRecords($this->parseIds($id), $attributes);
+        $elementIDList = array_pluck($insertData, $this->relatedPivotKey);
+
+        /**
+         * @event model.relation.beforeAttach
+         * Called before creating a new relation between models (only for BelongsToMany relation)
+         *
+         * Example usage:
+         *
+         *     $model->bindEvent('model.relation.beforeAttach', function ($relationName, $elementIDList, $insertData) use (\October\Rain\Database\Model $model) {
```

Comment:
Please use type hinting in the examples to let users know what type of argument they're getting, i.e. `(string) $relation, (array) $ids, (array) $data`